src/heap_item.rs

```diff
@@ -30,9 +30,6 @@ pub struct HeapItem {
 
 impl HeapItem {
     pub fn new(weight: Weight, node_id: NodeId) -> HeapItem {
-        if weight != weight {
```

Comment:
`usize` can't be NaN; I'm guessing this was leftover from a prototype when the weight was `f64`

Reply:
Yes, probably (no idea tbh).